miio/integrations/petfeeder/device.py

```diff
@@ -0,0 +1,313 @@
+            "Food storage lid: {result.food_storage_lid.name}\n"
+            "Days until cleaning required: {result.clean_days_left.days} day(s)\n"
+            "Days until desiccant replacement: {result.desiccant_days_left.days} day(s)\n"
+            "WiFi status led: {result.wifi_led}\n"
+            "Dispense button lock: {result.button_lock}\n",
+        )
+    )
+    def status(self) -> PetFeederStatus:
+        """Retrieve properties.
+
+        :return: Device attributes
+        :rtype: PetFeederStatus
+        """
+        properties = AVAILABLE_PROPERTIES.get(
+            self.model, AVAILABLE_PROPERTIES[MODEL_MMGG_FEEDER_PETFEEDER]
+        )
+        values = self.send("getprops")
+        return PetFeederStatus(defaultdict(lambda: None, zip(properties, values)))
+
+    @command(
+        click.argument("units", type=int),
+        default_output=format_output("Dispensing {units} unit(s)"),
+    )
+    def dispense_food(self, units: int) -> bool:
+        """Dispense food.
+
+        Manually dispense food.
+
+        :param int units: Number of units to dispense (1 unit ~= 5g)
+        :return: MiIO response
+        :rtype: bool
+        """
+        return self.send("outfood", [units]) == OK
+
+    @command(default_output=format_output("Resetting desiccant time left"))
+    def reset_desiccant_left(self) -> bool:
+        """Reset desiccant time.
+
+        Reset the desiccant life counter (30 days)
+
+        :return: MiIO response
+        :rtype: bool
+        """
+        date = (datetime.now() + timedelta(days=30)).strftime("%Y-%m-%d")
+        return self.send("resetdryer", [date]) == OK
+
+    @command(
+        click.argument("state", type=int),
+        default_output=format_output(
+            lambda state: "Turning on WiFi led" if state else "Turning off WiFi led"
+        ),
+    )
+    def set_wifi_led(self, state: bool) -> bool:
+        """Set wifi led.
+
+        Enable / Disable the wifi status led.
+
+        :param bool state: On / Off
+        :return: MiIO response
+        :rtype: bool
+        """
+        return self.send("wifiledon", [int(state)]) == OK
+
+    @command(
+        click.argument("state", type=int),
+        default_output=format_output(
+            lambda state: "Locking dispense button"
+            if state
+            else "Unlocking dispense button"
+        ),
+    )
+    def set_button_lock(self, state: bool) -> bool:
+        """Set button lock.
+
+        Lock / Unlock the manual dispense button.
+
+        :param bool state: Unlocked / Locked
+        :return: MiIO response
+        :rtype: bool
+        """
+        return self.send("keylock", [int(state) ^ 1]) == OK
+
+    @command(
+        click.argument("state", type=int),
+        default_output=format_output(
+            lambda state: "Enabling automatic feeding schedule"
+            if state
+            else "Disabling automatic feeding schedule"
+        ),
+    )
+    def set_feed_state(self, state: bool) -> bool:
```

Comment:
`set_automatic_feeding`? Or `enable_automatic_feeding`? The methods should use names that make their purpose clear to anyone reading the name :-)

Reply:
Was debating just calling it "mode".
0 = Manual
1 = Automatic

Not against ```set_automatic_feeding_state```

As I've said a couple of times, I'm against something that signifies state in it's function name (i.e _enable/_disable, _on/_off)